### PR TITLE
kbnm installer for OSX

### PR DESCRIPTION
**Pull request by shazow**
This PR is just for review/discussion/brainstorming. cc @gabriel 

I'm trying to instrument an installer for OSX that does the equivalent of https://github.com/keybase/client/blob/master/go/kbnm/install_host.

### CODE REVIEW

File: go/install/install_osx.go [hunk 1]
```diff
@@ -557,6 +566,75 @@ func InstallKBFS(context Context, binPath string, force bool, timeout time.Durat
 	return nil
 }
 
+// Return where the NativeMessaging host manifest lives on this platform.
+func kbnmManifestPath() string {
+	// Find path to write JSON into, one of:
+	// "/Library/Google/Chrome/NativeMessagingHosts"
+	// "$HOME/Library/Application Support/Google/Chrome/NativeMessagingHosts"
+	// XXX: Are we always root?
+	return "/Library/Google/Chrome/NativeMessagingHosts"
```
Comment (gabriel) [inline go/install/install_osx.go]: Please use the "$HOME/Library/Application Support/Google/Chrome/NativeMessagingHosts" dir?

Also NativeMessagingHosts dir doesn't exist for me there, so write would probably fail on my machine

File: go/install/install_osx.go [hunk 2]
```diff
@@ -557,6 +566,75 @@ func InstallKBFS(context Context, binPath string, force bool, timeout time.Durat
...
+		Type:        "stdio",
+		AllowedOrigins: []string{
+			kbnmExtensionID,
+		},
+	}
+
+	hostsPath := kbnmManifestPath()
+	jsonPath := filepath.Join(hostsPath, kbnmHostName+".json")
+
+	fp, err := os.OpenFile(jsonPath, os.O_WRONLY, 0444)
```
Comment (gabriel) [inline go/install/install_osx.go]: curious about 0444? assuming this is on purpose?

File: go/install/install_osx.go [hunk 3]
```diff
@@ -557,6 +566,75 @@ func InstallKBFS(context Context, binPath string, force bool, timeout time.Durat
...
+	return encoder.Encode(&hostManifest)
+}
+
+// UninstallKBNM removes the Keybase NativeMessaging whitelist
+func UninstallKBNM(log Log) error {
+	hostsPath := kbnmManifestPath()
+	jsonPath := filepath.Join(hostsPath, kbnmHostName+".json")
+
+	err := os.Remove(jsonPath)
+	if err != nil && !os.IsNotExist(err) {
```
Comment (gabriel) [inline go/install/install_osx.go]: you can combine this:

```
if err := os.Remove(jsonPath); err != nil && !os.IsNotExist(err) {
}
```

File: go/install/install_osx.go (hunk 1 above)
Comment (shazow) [inline go/install/install_osx.go]: Good point, need to mkdir it if it's not there.

File: go/install/install_osx.go (hunk 2 above)
Comment (shazow) [inline go/install/install_osx.go]: 0644 would work too. Actually now that I read it, I need `os.O_CREATE | os.O_WRONLY`.

Comment (gabriel) [inline go/install/install_osx.go]: yeah if there is anything secret in there you can do 0600 to be safe

Comment (shazow) [inline go/install/install_osx.go]: Not secret; 0444 is read-only for all.

**Comment by shazow** [comment]
@gabriel: Please take another look at this when you have a chance. Also any suggestions for how to best run/test/verify these things?

File: go/install/install_osx.go [hunk 4]
```diff
@@ -557,6 +567,96 @@ func InstallKBFS(context Context, binPath string, force bool, timeout time.Durat
 	return nil
 }
 
+// Return where the NativeMessaging host manifest lives on this platform.
+func kbnmManifestPath(u *user.User) string {
```
Comment (gabriel) [inline go/install/install_osx.go]: Go function comments are supposed to start with method name `// kbnmManifestPath returns where ...`

File: go/kbnm/main.go [hunk 5]
```diff
@@ -10,6 +10,9 @@ import (
 	"github.com/qrtz/nativemessaging"
 )
 
+// Version is the build version of kbnm, overwritten during build.
+const Version = "dev"
```
Comment (gabriel) [inline go/kbnm/main.go]: Make it real semver? 0.1.1  (< 1.0 is considered dev)

Comment (shazow) [inline go/kbnm/main.go]: This is to differentiate when it was built from `go build` vs through our script (which overrides the variable with ldflags). I think `0.1.1` might be not as obvious?

Comment (gabriel) [inline go/kbnm/main.go]: Oh cool, yeah setting through ldflag is better nm

**Comment by keybase-ci-visdiff** [comment]
The commits bf6be4d12a11...a8054ba2122d introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Code_Page-Show_Code_Mobile** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490801739409-bf6be4d12a11-a8054ba2122d-linux/Code_Page-Show_Code_Mobile-bf6be4d12a11.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490801739409-bf6be4d12a11-a8054ba2122d-linux/Code_Page-Show_Code_Mobile-a8054ba2122d.png)  
   Code_Page-Show_Code_Mobile comparison
 * Changed: **Code_Page-Show_Code** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490801739409-bf6be4d12a11-a8054ba2122d-linux/Code_Page-Show_Code-bf6be4d12a11.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490801739409-bf6be4d12a11-a8054ba2122d-linux/Code_Page-Show_Code-a8054ba2122d.png)  
   Code_Page-Show_Code comparison
</summary>

**Pull request merged**